pkg/api/vm/handler.go

```diff
@@ -595,30 +596,82 @@ func (h *vmActionHandler) addVolume(ctx context.Context, namespace, name string,
 	}
 
 	// Ref: https://kubevirt.io/api-reference/v0.44.0/operations.html#_v1vm-addvolume
-	return h.virtSubresourceRestClient.
+	if err = h.virtSubresourceRestClient.
 		Put().
 		Namespace(namespace).
 		Resource(vmResource).
 		Name(name).
 		SubResource(strings.ToLower(addVolume)).
 		Body(body).
 		Do(ctx).
-		Error()
+		Error(); err != nil {
+		return err
+	}
+
+	vm, err := h.vmCache.Get(namespace, name)
+	if err != nil {
+		return fmt.Errorf("failed to get vm %s/%s, error: %v", namespace, name, err)
+	}
+
+	var volumeClaimTemplates []corev1.PersistentVolumeClaim
+	anno := vm.GetAnnotations()
+	if volumeClaimTemplatesJSON, ok := anno[util.AnnotationVolumeClaimTemplates]; ok {
+		if err = json.Unmarshal([]byte(volumeClaimTemplatesJSON), &volumeClaimTemplates); err != nil {
+			return fmt.Errorf("failed to unserialize %s, error: %v", util.AnnotationVolumeClaimTemplates, err)
+		}
+	}
+
+	needAddVolume := true
+	for _, volumeClaimTemplate := range volumeClaimTemplates {
+		if volumeClaimTemplate.Name == input.VolumeSourceName {
+			needAddVolume = false
+		}
+	}
+	if needAddVolume {
+		volumeClaimTemplates = append(volumeClaimTemplates, corev1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      pvc.Name,
+				Namespace: pvc.Namespace,
```

Comment:
The namespace can be ignored here as it will always use the VM's by default.

Reply:
Removed it. Thanks.